lib/Service/FormsService.php

```diff
@@ -333,15 +333,12 @@ public function canSubmit(Form $form): bool {
 		if ($this->currentUser->getUID() === $form->getOwnerId()) {
 			return true;
 		}
-
 		// Refuse access, if SubmitMultiple is not set and user already has taken part.
-		if (!$form->getSubmitMultiple()) {
-			$participants = $this->submissionMapper->findParticipantsByForm($form->getId());
-			foreach ($participants as $participant) {
-				if ($participant === $this->currentUser->getUID()) {
-					return false;
-				}
-			}
+		if (
+			!$form->getSubmitMultiple()
+			&& $this->submissionMapper->hasFormSubmissionsByUser($form, $this->currentUser->getUID())
```

Comment:
I'd put the `&&` in the first line

```suggestion
			!$form->getSubmitMultiple() &&
            $this->submissionMapper->hasFormSubmissionsByUser($form, $this->currentUser->getUID())
```